torch/_higher_order_ops/templated_attention.py

```diff
@@ -0,0 +1,182 @@
+from typing import Callable
+
+import torch
+import torch.utils._pytree as pytree
+from torch._C import DispatchKey
+from torch._higher_order_ops.utils import (
+    _has_potential_branch_input_mutation,
+    autograd_not_implemented,
+    UnsupportedAliasMutationException,
+)
+from torch._ops import HigherOrderOperator
+from torch._subclasses import FakeTensorMode
+from torch.fx.experimental.proxy_tensor import (
+    make_fx,
+    ProxyTorchDispatchMode,
+    track_tensor_tree,
+)
+
+templated_attention = HigherOrderOperator("templated_attention")
```

Comment:
You should subclass HigherOrderOperator and override `__call__`. This way, we can do validation of the inputs and we can also annotate `__call__` with a docstring

Reply:
I kinda did this by exposing a different frontend, hiding the other buffers in torch.nn.attention  but Its probably good to document in multiple places